packages/core/src/StateMachine.ts

```diff
@@ -291,30 +312,22 @@ export class StateMachine<
    * entering the initial state.
    */
   public get initialState(): State<TContext, TEvent, TResolvedTypesMeta> {
-    const nextState = resolveMicroTransition(
-      this,
-      [],
-      this.preInitialState,
-      undefined
-    );
-    return macrostep(nextState, null as any, this);
+    return this.getInitialState();
   }
 
   /**
    * Returns the initial `State` instance, with reference to `self` as an `ActorRef`.
    */
-  public getInitialState(): State<TContext, TEvent, TResolvedTypesMeta> {
+  public getInitialState(): StateFrom<typeof this> {
+    const { preInitialState } = this;
     const nextState = resolveMicroTransition(
       this,
       [],
-      this.preInitialState,
+      preInitialState,
       undefined
     );
-    return macrostep(nextState, null as any, this) as State<
-      TContext,
-      TEvent,
-      TResolvedTypesMeta
-    >;
+    nextState.actions.unshift(...preInitialState.actions);
```

Comment:
did we miss some actions before? do we have a new test verifying this change?

Reply:
This is implicitly tested by numerous tests:

```
 FAIL  packages/core/test/actions.test.ts (14.654 s)
  ● sendTo › should be able to send an event to an actor

    : Timeout - Async callback was not invoked within the 5000 ms timeout specified by jest.setTimeout.Timeout - Async callback was not invoked within the 5000 ms timeout specified by jest.setTimeout.Error:

      1733 | 
      1734 | describe('sendTo', () => {
    > 1735 |   it('should be able to send an event to an actor', (done) => {
           |   ^
      1736 |     const childMachine = createMachine<any, { type: 'EVENT' }>({
      1737 |       initial: 'waiting',
      1738 |       states: {

      at new Spec (node_modules/@jest/core/node_modules/jest-jasmine2/build/jasmine/Spec.js:116:22)
      at Suite.<anonymous> (packages/core/test/actions.test.ts:1735:3)
      at Object.<anonymous> (packages/core/test/actions.test.ts:1734:1)

  ● sendTo › should be able to send an event from expression to an actor

    : Timeout - Async callback was not invoked within the 5000 ms timeout specified by jest.setTimeout.Timeout - Async callback was not invoked within the 5000 ms timeout specified by jest.setTimeout.Error:

      1758 |   });
      1759 | 
    > 1760 |   it('should be able to send an event from expression to an actor', (done) => {
           |   ^
      1761 |     const childMachine = createMachine<any, { type: 'EVENT'; count: number }>({
      1762 |       initial: 'waiting',
      1763 |       states: {

      at new Spec (node_modules/@jest/core/node_modules/jest-jasmine2/build/jasmine/Spec.js:116:22)
      at Suite.<anonymous> (packages/core/test/actions.test.ts:1760:3)
      at Object.<anonymous> (packages/core/test/actions.test.ts:1734:1)

 FAIL  packages/core/test/actor.test.ts (20.186 s)
  ● actors › should only spawn an initial actor once when it synchronously responds with an event

    expect(received).toEqual(expected) // deep equality

    Expected: "done"
    Received: "testing"

      715 | 
      716 |     const service = interpret(testMachine).start();
    > 717 |     expect(service.state.value).toEqual('done');
          |                                 ^
      718 |   });
      719 | 
      720 |   it('should spawn null actors if not used within a service', () => {

      at Object.<anonymous> (packages/core/test/actor.test.ts:717:33)

  ● actors › with behaviors › should work with a promise behavior (reject)

    : Timeout - Async callback was not invoked within the 5000 ms timeout specified by jest.setTimeout.Timeout - Async callback was not invoked within the 5000 ms timeout specified by jest.setTimeout.Error:

      1207 |     });
      1208 | 
    > 1209 |     it('should work with a promise behavior (reject)', (done) => {
           |     ^
      1210 |       const errorMessage = 'An error occurred';
      1211 |       const countMachine = createMachine<{
      1212 |         count: ActorRefFrom<Promise<number>>;

      at new Spec (node_modules/@jest/core/node_modules/jest-jasmine2/build/jasmine/Spec.js:116:22)
      at Suite.<anonymous> (packages/core/test/actor.test.ts:1209:5)
      at Suite.<anonymous> (packages/core/test/actor.test.ts:1129:3)
      at Object.<anonymous> (packages/core/test/actor.test.ts:601:1)

  ● actors › should be able to spawn callback actors in (lazy) initial context

    : Timeout - Async callback was not invoked within the 5000 ms timeout specified by jest.setTimeout.Timeout - Async callback was not invoked within the 5000 ms timeout specified by jest.setTimeout.Error:

      1293 |   });
      1294 | 
    > 1295 |   it('should be able to spawn callback actors in (lazy) initial context', (done) => {
           |   ^
      1296 |     const machine = createMachine<{ ref: ActorRef<any> }>({
      1297 |       context: ({ spawn }) => ({
      1298 |         ref: spawn(

      at new Spec (node_modules/@jest/core/node_modules/jest-jasmine2/build/jasmine/Spec.js:116:22)
      at Suite.<anonymous> (packages/core/test/actor.test.ts:1295:3)
      at Object.<anonymous> (packages/core/test/actor.test.ts:601:1)

  ● actors › should be able to spawn machines in (lazy) initial context

    : Timeout - Async callback was not invoked within the 5000 ms timeout specified by jest.setTimeout.Timeout - Async callback was not invoked within the 5000 ms timeout specified by jest.setTimeout.Error:

      1320 |   });
      1321 | 
    > 1322 |   it('should be able to spawn machines in (lazy) initial context', (done) => {
           |   ^
      1323 |     const childMachine = createMachine({
      1324 |       entry: sendParent('TEST')
      1325 |     });

      at new Spec (node_modules/@jest/core/node_modules/jest-jasmine2/build/jasmine/Spec.js:116:22)
      at Suite.<anonymous> (packages/core/test/actor.test.ts:1322:3)
      at Object.<anonymous> (packages/core/test/actor.test.ts:601:1)

Summary of all failing tests
 FAIL  packages/core/test/actions.test.ts (14.654 s)
  ● sendTo › should be able to send an event to an actor

    : Timeout - Async callback was not invoked within the 5000 ms timeout specified by jest.setTimeout.Timeout - Async callback was not invoked within the 5000 ms timeout specified by jest.setTimeout.Error:

      1733 | 
      1734 | describe('sendTo', () => {
    > 1735 |   it('should be able to send an event to an actor', (done) => {
           |   ^
      1736 |     const childMachine = createMachine<any, { type: 'EVENT' }>({
      1737 |       initial: 'waiting',
      1738 |       states: {

      at new Spec (node_modules/@jest/core/node_modules/jest-jasmine2/build/jasmine/Spec.js:116:22)
      at Suite.<anonymous> (packages/core/test/actions.test.ts:1735:3)
      at Object.<anonymous> (packages/core/test/actions.test.ts:1734:1)

  ● sendTo › should be able to send an event from expression to an actor

    : Timeout - Async callback was not invoked within the 5000 ms timeout specified by jest.setTimeout.Timeout - Async callback was not invoked within the 5000 ms timeout specified by jest.setTimeout.Error:

      1758 |   });
      1759 | 
    > 1760 |   it('should be able to send an event from expression to an actor', (done) => {
           |   ^
      1761 |     const childMachine = createMachine<any, { type: 'EVENT'; count: number }>({
      1762 |       initial: 'waiting',
      1763 |       states: {

      at new Spec (node_modules/@jest/core/node_modules/jest-jasmine2/build/jasmine/Spec.js:116:22)
      at Suite.<anonymous> (packages/core/test/actions.test.ts:1760:3)
      at Object.<anonymous> (packages/core/test/actions.test.ts:1734:1)

 FAIL  packages/core/test/actor.test.ts (20.186 s)
  ● actors › should only spawn an initial actor once when it synchronously responds with an event

    expect(received).toEqual(expected) // deep equality

    Expected: "done"
    Received: "testing"

      715 | 
      716 |     const service = interpret(testMachine).start();
    > 717 |     expect(service.state.value).toEqual('done');
          |                                 ^
      718 |   });
      719 | 
      720 |   it('should spawn null actors if not used within a service', () => {

      at Object.<anonymous> (packages/core/test/actor.test.ts:717:33)

  ● actors › with behaviors › should work with a promise behavior (reject)

    : Timeout - Async callback was not invoked within the 5000 ms timeout specified by jest.setTimeout.Timeout - Async callback was not invoked within the 5000 ms timeout specified by jest.setTimeout.Error:

      1207 |     });
      1208 | 
    > 1209 |     it('should work with a promise behavior (reject)', (done) => {
           |     ^
      1210 |       const errorMessage = 'An error occurred';
      1211 |       const countMachine = createMachine<{
      1212 |         count: ActorRefFrom<Promise<number>>;

      at new Spec (node_modules/@jest/core/node_modules/jest-jasmine2/build/jasmine/Spec.js:116:22)
      at Suite.<anonymous> (packages/core/test/actor.test.ts:1209:5)
      at Suite.<anonymous> (packages/core/test/actor.test.ts:1129:3)
      at Object.<anonymous> (packages/core/test/actor.test.ts:601:1)

  ● actors › should be able to spawn callback actors in (lazy) initial context

    : Timeout - Async callback was not invoked within the 5000 ms timeout specified by jest.setTimeout.Timeout - Async callback was not invoked within the 5000 ms timeout specified by jest.setTimeout.Error:

      1293 |   });
      1294 | 
    > 1295 |   it('should be able to spawn callback actors in (lazy) initial context', (done) => {
           |   ^
      1296 |     const machine = createMachine<{ ref: ActorRef<any> }>({
      1297 |       context: ({ spawn }) => ({
      1298 |         ref: spawn(

      at new Spec (node_modules/@jest/core/node_modules/jest-jasmine2/build/jasmine/Spec.js:116:22)
      at Suite.<anonymous> (packages/core/test/actor.test.ts:1295:3)
      at Object.<anonymous> (packages/core/test/actor.test.ts:601:1)

  ● actors › should be able to spawn machines in (lazy) initial context

    : Timeout - Async callback was not invoked within the 5000 ms timeout specified by jest.setTimeout.Timeout - Async callback was not invoked within the 5000 ms timeout specified by jest.setTimeout.Error:

      1320 |   });
      1321 | 
    > 1322 |   it('should be able to spawn machines in (lazy) initial context', (done) => {
           |   ^
      1323 |     const childMachine = createMachine({
      1324 |       entry: sendParent('TEST')
      1325 |     });

      at new Spec (node_modules/@jest/core/node_modules/jest-jasmine2/build/jasmine/Spec.js:116:22)
      at Suite.<anonymous> (packages/core/test/actor.test.ts:1322:3)
      at Object.<anonymous> (packages/core/test/actor.test.ts:601:1)
```